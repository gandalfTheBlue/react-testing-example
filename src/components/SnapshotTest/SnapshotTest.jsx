import axios from 'axios'
import React, { useState } from 'react'
import { useEffect } from 'react'

//https://dev.to/nicomartin/how-to-test-async-react-hooks-392j

const SnapshotTest = () => {
  const [data, setData] = useState()

  useEffect(() => {
    const fetchData = async () => {
      const result = await axios.get(
        `https://api.github.com/repos/ant-design/ant-design`
      )
      setData(result.data)
    }
    fetchData()
  }, [])

  if (!data) return <div>loading...</div>

  return <div>{data.name}</div>
}

export default SnapshotTest
