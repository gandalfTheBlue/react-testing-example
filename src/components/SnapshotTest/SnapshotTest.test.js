import React from 'react'
import renderer from 'react-test-renderer'
import SnapshotTest from './SnapshotTest'
import axios from 'axios'
const { act } = renderer

jest.mock('axios')

describe('<SnapshotTest/>', () => {
  beforeEach(() => {
    axios.get.mockImplementationOnce(() =>
      Promise.resolve({ data: { name: '3333' } })
    )
  })

  it('renders correctly SnapshotTest', () => {
    const tree = renderer.create(<SnapshotTest />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly SnapshotTest after fetch data', async () => {
    const tree = renderer.create(<SnapshotTest />)
    await act(() => Promise.resolve())
    await expect(tree.toJSON()).toMatchSnapshot()
  })
})
