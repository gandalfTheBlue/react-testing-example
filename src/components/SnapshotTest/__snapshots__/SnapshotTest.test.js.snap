// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnapshotTest/> renders correctly SnapshotTest 1`] = `
<div>
  loading...
</div>
`;

exports[`<SnapshotTest/> renders correctly SnapshotTest after fetch data 1`] = `
<div>
  3333
</div>
`;
